packages/codecs-data-structures/src/map.ts

```diff
@@ -87,7 +87,7 @@ export function getMapDecoder<TToKey, TToValue>(
     config: MapCodecConfig<NumberDecoder> = {},
 ): Decoder<Map<TToKey, TToValue>> {
     return mapDecoder(
-        getArrayDecoder(getTupleDecoder([key, value]), config as object),
+        getArrayDecoder(getTupleDecoder([key, value]), config as object) as Decoder<[TToKey, TToValue][]>,
```

Comment:
That's only because of the usual function overload issue that causes TypeScript to not know which overload to choose. With this cast, I internally tell it to choose one of the overload instead of adding a condition that end up in writing the same code twice.